### ISSUE TITLE
Improve Homepage alert banner publishing process

### ISSUE THREAD

**Issue by mchelen-gov** [department-of-veterans-affairs/va.gov-team#26991]
## High Level User Story/ies
As a content editor, I need to quickly publish alerts on the VA.gov homepage, so that time sensitive information can reach Veterans.

## Hypothesis or Bet
**If** _we improve the process for publishing updates to homepage alert_ **then** _we lower coordination work and more quickly and reliably deliver content to Veterans_.

## OKR
_Which Objective / Key Result does this epic push forward?_

- Self service
- Reliability

## Existing process
- PR is opened in `vagov-content` i.e. https://github.com/department-of-veterans-affairs/vagov-content/pull/776
  - YAML is updated with HTML snipped
- PR is merged
- automated content only build is triggered hourly during business hours
  - [Builds / content-build content only build for vagovprod](http://jenkins.vfs.va.gov/job/builds/job/content-build-content-only-vagovprod)
- manual content-only build is triggered using jenkins
  - [Deploys / Vets.gov Auto-Deploy for content-build](http://jenkins.vfs.va.gov/blue/organizations/jenkins/deploys%2Fvets-gov-autodeploy-content-build/activity)

### Add HP banner content to CMS
- pros
    - does not require OOB deploy
    - uses existing content publishing pipeline
- cons
    - requires CMS/GFE access
    - requires CMS permissions and field
    - can be blocked by content build errors
- mitigations
    - improve specificity of permissions
    - have OCTODE staff perform content updates
    
### Load data from public Github repo using React widget
- pros
    - does not require GFE access
    - does not depend on content build 
- cons
    - malformed data could break homepage
    - adds additional source for content on va.gov outside of content build
    - handling of different environments (dev/stage/prod) unclear
- mitigations
    - add CI/CD tests to repo

### What must be true in order for you to consider this epic complete? 

*Take into consideration Accessibility/QA needs as well as Product, Technical, and Design requirements.*

## How to configure this issue
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `tools-be`, `tools-fe`,`content-ia`, `service-design`, `vsa-*`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)

**Comment by mchelen-gov** [comment]
Related OOB PM: https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Postmortems/2021-07-06-homepage-banner-update.md

**Comment by kevwalsh** [comment]
Today @swirtSJW  and i briefly discussed what an iteration on the content model and governance structure for the existing banner alert structure might look like (which was never really meant to scale, it was a barebones implementation to accomplish the Benefits Hubs migration). 

1. A new content type called, say, **Banner**, modelled after VAMC banner alert but without the facility-specific aspects. 
2. New banners would be created through some kind of request fulfilment, processed by, say, the sitewide content team. Existing banners like the homepage banner would not need additional requests, editors would be able to archive, then republish them later, when needed. Hard to know how many requests would come in, i would expect that most new banners are created and managed internal to the sitewide content team anyway. 
3. A "scope" field (field name TBD) that can only be edited by content admins, as part of the request fulfilment. It would target URL paths or path patterns on va.gov (including paths that are not nodes, or not even known to Drupal, which is a limitation of the existing alerts in CMS). So a banner request would include what URLs on which the banner should appear, and the sitewide content team would be the arbiter of that. This would remove the risk mentioned in slack of editors being able to add a banner to any node they want.  FE queries and templates would add the banner to pages during the build, based on these scope values.
4. Saving as published or as archived would trigger a content build, much like VAMC banner alerts. We could consider an additional boolean field (again, not editable by the end user) that would determine if saving as published or archived actually causes that trigger, since some banners may not be urgent. 
5. Section based content ownership, like any other piece of content managed in the CMS. So a block associated with OPIA would only be edited by OPIA (currently only Josh T, though i imagine others would be added), a block owned by VHA Digital Comms could only be edited by members of that section, etc. 
6. Any existing banner alerts managed by the existing "alert" block type would be migrated to use this architecture, reducing the latter block type's scope strictly to reusable in-page alert blocks, like the mental health alert block, which was its original purpose.

I'm sure this is incomplete and any number of these points need some more thinking. And nothing here addresses the concerns that banners as CMS content would depend on content build succeeding, although i think it's worth noting this risk is already present for VAMC banners.

**Comment by indytechcook** [comment]
One more possible use case

### Load data from JSON file on S3 populated from Drupal
- pros
   - does not require OOB deploy
   - does not require a full content release
   - alert i loaded in real time
- cons
  - requires CMS/GFE access
  - **requires a new publishing pipeline**

**Comment by johnhashva** [comment]
To put a finer point on a few things:

**Comment by johnhashva** [comment]
To put a finer point on a few things:

**Comment by johnhashva** [comment]
Additional context as we think this through:

- Banner must be deployed into production as quickly as possible -- 5-10 minutes.
- Optimally: A broken link checker must be in place to "force" the editor to fix those links before the banner can be published.
- Optimally: a spell checker is part of this solution

**Comment by jenniferlee-dsva** [comment]
Wanted to provide this reference which captures some of the early discussions that @johnhashva @davidconlon @rtwell and at the time someone on the Drupal engineering team (forget her name). 

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/banners/banner-alerts.md 

I think ^ @kevwalsh is including in-body alert components in # 6. That may be appropriate, but wanted to point that out, so we don't unintentionally conflate the full-width banners (and specifically the one that appears on the homepage only) with the in-body alert component. 

The homepage banners do not go through the content team. (The pandemic messaging is/was a special case bc of everything COVID related, even those 'owned' by OPIA, having to go through the Joint Task Force.)

I would add additionally that the outcome we'd like to see is for OPIA (Josh) to have the ability to publish HOMEPAGE ONLY banner message texts -- and not be able to publish them to any other nodes. (The sitewide  team should determine whether messaging belongs on other global pages.) 

Josh is going to need a draft preview ability for HP banners. BC just saving to publish WILL guarantee lots of mistakes going out and then having to re-do after publishing.

**Comment by mchelen-gov** [comment]
@kevwalsh How is the new content type you are describing different from the existing Alert content type (https://prod.cms.va.gov/block/add/alert?destination=/admin/content/alerts)?

**Comment by mchelen-gov** [comment]
@timwright12 has confirmed that `vagov-content` is used in Jenkins job [Builds / content-build content only build for vagovprod](http://jenkins.vfs.va.gov/job/builds/job/content-build-content-only-vagovprod)
> https://dsva.slack.com/archives/C01512KN35G/p1626722097258200?thread_ts=1626721392.255400&cid=C01512KN35G

**Comment by mchelen-gov** [comment]
@kevwalsh i believe HP banner was migrated to Drupal in another issue, can you confirm?

**Comment by FranECross** [comment]
@jilladams All of the tickets associated to this epic have been closed. Do you know if this epic can also be closed or if there is more work that needs to be done that wasn't ticketed? Thanks!

**Comment by jilladams** [comment]
Oo, closing!
